lib/modules/accountSwitcher.js

```diff
@@ -312,21 +317,24 @@ function findMatchingAccount(username, partialMatch) {
 async function switchTo(user) {
 	const account = findMatchingAccount(user);
 	if (!account) return;
-	const [username, password] = account;
+	const [username, password, uses2Fa] = account;
 
-	if (loggedInUser()) {
-		await ajax({
-			method: 'POST',
-			url: '/logout',
-		});
+	// Don't await promise yet so that the PIN can be entered while logging out
+	const logoutPromise = loggedInUser() && ajax({ method: 'POST', url: '/logout' });
+
+	let pin;
+	if (uses2Fa) {
+		pin = window.prompt(`Enter 2FA PIN for user ${user}`);
```

Comment:
`i18n("Enter the 6 digit code from your authenticator app for u/%s", username)`

(except properly formatted for RES)